src/briefcase/commands/create.py

```diff
@@ -672,6 +672,10 @@ def create_app(self, app: BaseConfig, **options):
         self.logger.info("Installing support package...", prefix=app.app_name)
         self.install_app_support_package(app=app)
 
+        # Verify tools for the app after the app template and support package
+        # are in place since the app tools may be dependent on them.
+        self.verify_app_tools(app)
+
```

Comment:
Right now, the verification of Docker includes calling `Docker.prepare()` (which is responsible for ensuring the Docker image exists and is updated via `docker build`). Calling `Docker.prepare()` is dependent on the existence of the app template...since the template includes `Dockerfile`. 

The verification of app-tools can happen sooner (which I also think makes the most sense) but we'll need to find a spot to run `docker build` before `docker run` is invoked.

Reply:
Ah, OK - that makes sense. And now that I see and actually *read* the comment, I see you explained that, too :-)